### PR TITLE
Add criteria to README and issue template

### PR DESCRIPTION
**Pull request by frankharkins**
### Summary

Adds a criteria for new submissions going forward. This will eventually apply to existing ecosystem projects but is not enforced yet.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,9 +32,11 @@ repository. We'll review the pull request and get back to you within a few
 days.
 
 To join the ecosystem, your project must:
-  - Have an open-source license
-  - Adhere to our [code of conduct](./CODE_OF_CONDUCT.md)
-  - Build on, interface with, or extend Qiskit in some way
+  - Build on, interface with, or extend the Qiskit SDK in some way.
```
Comment (Eric-Arellano) [inline README.md:35]: Maybe link to the Qiskit repo or docs?

File: README.md [hunk 2]
```diff
@@ -32,9 +32,11 @@ repository. We'll review the pull request and get back to you within a few
...
+  - Be compatible with Qiskit 1.0 (or newer).
+  - Have an [OSI-approved](https://opensource.org/license?categories=popular-strong-community) open-source license (preferably Apache 2.0 or MIT).
+  - Adhere to our [code of conduct](./CODE_OF_CONDUCT.md).
```
Comment (Eric-Arellano) [inline README.md:38]: Should we offer that you can have an additional code of conduct?

File: README.md [hunk 3]
```diff
@@ -32,9 +32,11 @@ repository. We'll review the pull request and get back to you within a few
...
+  - Have a commit from within the last 6 months.
```
Comment (Eric-Arellano) [inline README.md:39]:
```suggestion
  - Have maintainer activity within the last 6 months, such as a commit.
```

**Review by Eric-Arellano — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (frankharkins) [reply README.md:38]: You mean they can enforce their own in addition to ours?

Comment (Eric-Arellano) [reply README.md:38]: Exactly. Like theirs can be a superset.

**Review by Eric-Arellano — APPROVED** [review]

**Pull request merged**